### PR TITLE
#161779800 Add Role based access control functionality

### PR DESCRIPTION
**Pull request by jideajayi11**
#### What does this PR do?
* This PR updates user role and ensures that only admin can update a user

#### Description of Task to be completed?
* Create the model for roles
* Create seeder for the roles model
* Wrote tests for validators and controller
* Wrote validator method
* Wrote controller method

#### How should this be manually tested?
* PUT `/api/v1/users/role/:userId`
* Body {roleId}

#### Any background context you want to provide?
* roleId can only be `1, 2 or 3` as in the model

#### What are the relevant pivotal tracker stories?
* [ #161779800](https://www.pivotaltracker.com/story/show/161779800)

#### Screenshots (if appropriate)
* N/A

#### Questions:
* N/a

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,16 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "start": "nodemon ./src/app.js --exec babel-node",
-    "test": "npm run undo:migration && npm run migrate && export NODE_ENV=test&& nyc --reporter=html --reporter=text mocha --exit --require babel-register --require babel-polyfill ./test/index.js",
+    "start": "npm run undo:seeder && npm run seeder && nodemon ./src/app.js --exec babel-node",
```
Comment (kingisaac95) [inline package.json]: Remove seed script from start

File: package.json [hunk 2]
```diff
@@ -4,14 +4,16 @@
...
+    "test": "npm run undo:migration && npm run migrate && export NODE_ENV=test && npm run undo:seeder && npm run seeder && nyc --reporter=html --reporter=text mocha --exit --require babel-register --require babel-polyfill ./test/index.js",
```
Comment (kingisaac95) [inline package.json]: Move this `export NODE_ENV=test` to the start of the script

**Review by Daymorelah — COMMENTED** [review]
Good job @jideajayi11

**Review by kingisaac95 — APPROVED** [review]

**Comment by kingisaac95** [comment]
Good job! 💪

**Pull request merged**